### PR TITLE
Proposed README changeset

### LINKED ISSUE #450: Simplify Developer section of README
Developer install has some unnecessary lines, trim it down.  (Also example of using github for recent developers).

### PR DESCRIPTION
**Pull request by garrettwrong**
Minor README changes.  Has some typos for reviewers to catch.

Closes #450 .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,26 +41,26 @@ After cloning this repo, the simplest option is to use Anaconda 64-bit for your
...
+# Creates the conda environment and installs base dependencies.
 conda env create -f environment.yml --name aspire_dev
 
 # Enable the environment
 conda activate aspire_dev
 
-# Install the aspire package in a locally editable way:
-pip install -e .
+# Install the aspire package in a locally editable way,
+# and additional installs the developer tools extras:
```
Comment (chris-langfield) [inline README.md]: and additionally installs

**Review by chris-langfield — CHANGES_REQUESTED** [review]
grammar

File: README.md [hunk 2]
```diff
@@ -41,26 +41,26 @@ After cloning this repo, the simplest option is to use Anaconda 64-bit for your
...
+pip install -e ".[dev]"
+
 ```
 
 If you prefer not to use Anaconda, or want to manage environments yourself, you should be able to use `pip` with Python >= 3.6.
 Please see the full documentation for details.
 
-You may optionally install additional packages:
+You may optionally install additional packages for GPU extension:
```
Comment (j-c-c) [inline README.md]: I think you meant "extensions"

**Review by j-c-c — CHANGES_REQUESTED** [review]
Looks good other than typo

File: README.md (hunk 1 above)
Comment (garrettwrong) [reply README.md]: yup, fixed.

File: README.md (hunk 2 above)
Comment (garrettwrong) [reply README.md]: indeed, thanks

**Review by chris-langfield — APPROVED** [review]
nice!

**Review by j-c-c — APPROVED** [review]

**Review by janden — APPROVED** [review]

**Pull request merged**